Sources/NIO/BaseSocket.swift

```diff
@@ -319,6 +327,7 @@ class BaseSocket: Selectable {
     init(descriptor: CInt) {
         precondition(descriptor >= 0, "invalid file descriptor")
         self.descriptor = descriptor
+        try! ignoreSIGPIPE(descriptor: descriptor)
```

Comment:
Isn't the dance with the atomic above unnecessary? Just do it inside an implicit `dispatch_once` through means of a global or static let.

Reply:
no, thanks